datasetinsights/estimators/faster_rcnn.py

```diff
@@ -22,6 +22,594 @@
+                                       dataset=train_dataset,
+                                       is_train=True)
+        val_sampler = create_sampler(is_distributed=is_distributed,
+                                     dataset=val_dataset,
+                                     is_train=False)
+
+        train_loader = create_dataloader(config, train_dataset, train_sampler,
+                                         TRAIN)
+        val_loader = create_dataloader(config, val_dataset, val_sampler,
+                                       VAL)
+        self.train_loop(
+            train_dataloader=train_loader,
+            label_mappings=label_mappings,
+            val_dataloader=val_loader,
+            train_sampler=train_sampler,
+        )
+
+    def train_loop(
+        self,
+        *,
+        train_dataloader,
+        label_mappings,
+        val_dataloader,
+        train_sampler=None,
+    ):
+        """
+        Args:
+            train_dataloader (torch.utils.data.DataLoader):
+            label_mappings (dict): a dict of {label_id: label_name} mapping
+            val_dataloader (torch.utils.data.DataLoader)
+            train_sampler: https://pytorch.org/docs/stable/data.html#torch.utils.data.Sampler
+        """
+        model = self.model.to(self.device)
+        optimizer, lr_scheduler = create_optimizer(self.config, model)
+        accumulation_steps = self.config.train.get(
+            "accumulation_steps", DEFAULT_ACCUMULATION_STEPS
+        )
+        logger.debug("Start training")
+        total_timer = Timer(
+            name="total-time", text=const.TIMING_TEXT, logger=logging.info
+        )
+        total_timer.start()
+        for epoch in range(self.config.train.epochs):
+            with Timer(
+                name=f"epoch-{epoch}-train-time",
+                text=const.TIMING_TEXT,
+                logger=logging.info,
+            ):
+                self.train_one_epoch(
+                    optimizer=optimizer,
+                    data_loader=train_dataloader,
+                    epoch=epoch,
+                    lr_scheduler=lr_scheduler,
+                    accumulation_steps=accumulation_steps,
+                )
+            if self.config.system.distributed:
+                train_sampler.set_epoch(epoch)
+            self.checkpointer.save(self, epoch=epoch)
+            with Timer(
+                name=f"epoch-{epoch}-evaluate-time",
+                text=const.TIMING_TEXT,
+                logger=logging.info,
+            ):
+                self.evaluate_per_epoch(
+                    data_loader=val_dataloader,
+                    epoch=epoch,
+                    label_mappings=label_mappings,
+                )
+        total_timer.stop()
+
+    def train_one_epoch(
+            self,
+            *,
+            optimizer,
+            data_loader,
+            epoch,
+            lr_scheduler,
+            accumulation_steps,
+    ):
+        """
+
```

Comment:
Improve docstring here. It needs to have title and description about this public method.

Reply:
Fixed